csaf/src/source/http.rs

```diff
@@ -84,11 +91,38 @@ impl Source for HttpSource {
     type Error = HttpSourceError;
 
     async fn load_metadata(&self) -> Result<ProviderMetadata, Self::Error> {
-        Ok(self
-            .fetcher
-            .fetch::<Json<ProviderMetadata>>(self.url.clone())
-            .await?
-            .into_inner())
+        let metadata_retriever = MetadataRetriever {
+            base_url: self.url.clone(),
+            fetcher: self.fetcher.clone(),
+        };
+
+        if self.url.contains(WELL_KNOWN_METADATA) {
```

Comment:
I am not sure why we need this?

Reply:
as we discussed, if the user inputs full well known URL, we do not needs to do discovery chain.

Reply:
We discussed that if the user enters a URL we will just accept it as target for the provider metadata. There is no need to check for it to "contain" or "end with" the well known part. If it's a `https://` URL, we just accept it.

Reply:
OK I will remove it.